libraries/AC_PID/AC_PID.cpp

```diff
@@ -126,6 +153,37 @@ void AC_PID::slew_limit(float smax)
     _slew_rate_max.set(fabsf(smax));
 }
 
+void AC_PID::set_notch_sample_rate(float sample_rate)
+{
+#if AP_FILTER_ENABLED
+    if (_notch_T_filter == 0 && _notch_E_filter == 0) {
+        return;
+    }
+
+    if (_notch_T_filter != 0) {
+        if (_target_notch == nullptr) {
+            _target_notch = new NotchFilterFloat();
+        }
+        AP_Filter* filter = AP::filters().get_filter(_notch_T_filter);
+        if (filter != nullptr && !filter->setup_notch_filter(*_target_notch, sample_rate)) {
+            delete _target_notch;
```

Comment:
we need to set _target_notch to nullptr here or we could use deleted memory

Reply:
Fixed